include/seahorn/Expr/ExprOpBv.hh

```diff
@@ -165,59 +179,188 @@ enum class BvOpKind {
-NOP(BSMOD, "bvsmod", FUNCTIONAL, BvOp)
-NOP(BULT, "bvult", FUNCTIONAL, BvOp)
-NOP(BSLT, "bvslt", FUNCTIONAL, BvOp)
-NOP(BULE, "bvule", FUNCTIONAL, BvOp)
-NOP(BSLE, "bvsle", FUNCTIONAL, BvOp)
-NOP(BUGE, "bvuge", FUNCTIONAL, BvOp)
-NOP(BSGE, "bvsge", FUNCTIONAL, BvOp)
-NOP(BUGT, "bvugt", FUNCTIONAL, BvOp)
-NOP(BSGT, "bvsgt", FUNCTIONAL, BvOp)
-NOP(BCONCAT, "concat", FUNCTIONAL, BvOp)
-NOP(BEXTRACT, "extract", FUNCTIONAL, BvOp)
-NOP(BSEXT, "bvsext", FUNCTIONAL, BvOp)
-NOP(BZEXT, "bvzext", FUNCTIONAL, BvOp)
-NOP(BREPEAT, "bvrepeat", FUNCTIONAL, BvOp)
-NOP(BSHL, "bvshl", FUNCTIONAL, BvOp)
-NOP(BLSHR, "bvlshr", FUNCTIONAL, BvOp)
-NOP(BASHR, "bvashr", FUNCTIONAL, BvOp)
-NOP(BROTATE_LEFT, "bvrotleft", FUNCTIONAL, BvOp)
-NOP(BROTATE_RIGHT, "bvrotright", FUNCTIONAL, BvOp)
-NOP(BEXT_ROTATE_LEFT, "bvextrotleft", FUNCTIONAL, BvOp)
-NOP(BEXT_ROTATE_RIGHT, "bvextrotright", FUNCTIONAL, BvOp)
-NOP(INT2BV, "int2bv", FUNCTIONAL, BvOp)
-NOP(BV2INT, "bv2int", FUNCTIONAL, BvOp)
+NOP_TYPECHECK(BNOT, "bvnot", FUNCTIONAL, BvOp, typeCheck::bvType::Unary)
+NOP_TYPECHECK(BREDAND, "bvredand", FUNCTIONAL, BvOp, typeCheck::bvType::Unary)
+NOP_TYPECHECK(BREDOR, "bvredor", FUNCTIONAL, BvOp, typeCheck::bvType::Unary)
+NOP_TYPECHECK(BAND, "bvand", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BOR, "bvor", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BXOR, "bvxor", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BNAND, "bvnand", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BNOR, "bvnor", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BXNOR, "bvxnor", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BNEG, "bvneg", FUNCTIONAL, BvOp, typeCheck::bvType::Unary)
+NOP_TYPECHECK(BADD, "bvadd", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSUB, "bvsub", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BMUL, "bvmul", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BUDIV, "bvudiv", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSDIV, "bvsdiv", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BUREM, "bvurem", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSREM, "bvsrem", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSMOD, "bvsmod", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BULT, "bvult", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSLT, "bvslt", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BULE, "bvule", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSLE, "bvsle", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BUGE, "bvuge", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSGE, "bvsge", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BUGT, "bvugt", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BSGT, "bvsgt", FUNCTIONAL, BvOp, typeCheck::bvType::Nary)
+NOP_TYPECHECK(BCONCAT, "concat", FUNCTIONAL, BvOp, typeCheck::bvType::Concat)
```

Comment:
concat can be nary -- did not check whether this is supported, but ran into this problem before so mentioning it here